http-client.ts

```diff
@@ -40,17 +42,23 @@ export class HttpClient implements Server.IHttpClient {
 		let pipeTo = options.pipeTo;
 		delete options.pipeTo;
 
-		let proto = this.$config.USE_PROXY ? "http" : requestProto;
+		const proxyCache = this.$proxyService.getCache();
+		let proto = proxyCache && proxyCache.USE_PROXY ? "http" : requestProto;
 		let http = require(proto);
 
 		options.headers = options.headers || {};
 		let headers = options.headers;
 
-		if (proxySettings || this.$config.USE_PROXY) {
+		if (proxySettings || (proxyCache && proxyCache.USE_PROXY)) {
+			const proxyCredentials = await this.$proxyService.getCredentials();
 			options.path = requestProto + "://" + options.host + options.path;
 			headers.Host = options.host;
-			options.host = (proxySettings && proxySettings.hostname) || this.$config.PROXY_HOSTNAME;
-			options.port = (proxySettings && proxySettings.port) || this.$config.PROXY_PORT;
+			options.host = (proxySettings && proxySettings.hostname) || proxyCache.PROXY_HOSTNAME;
+			options.port = (proxySettings && proxySettings.port) || proxyCache.PROXY_PORT;
+			if (proxyCredentials && proxyCredentials.username && proxyCredentials.password) {
```

Comment:
I would move the `const proxyCredentials` just above the if, as it's the first & only place it's used and it will be easier to read for me. There are too many proxy* variables anyway.